hathor/transaction/transaction_metadata.py

```diff
@@ -129,7 +133,7 @@ def __init__(
 
         self.feature_activation_bit_counts = feature_activation_bit_counts
 
-        settings = get_global_settings()
+        settings = settings or get_global_settings()
 
         # Genesis specific:
         if hash is not None and is_genesis(hash, settings=settings):
```

Comment:
Is this the only use for settings? Should we refactor this `__init__()` method to receive a `is_genesis: bool` attribute (instead of settings)? Or check with the storage?

Reply:
Similarly to what I described in https://github.com/HathorNetwork/hathor-core/pull/1085#discussion_r1679874120, I don't think there's any advantage in changing this. There's no benefit in leaving the responsibility of passing the correct value of `is_genesis` to the caller, as this is an internal logic to this class. About the storage, it's not available in this context, so it would also have to be passed as an argument anyway. So it's better to just pass the settings.

Reply:
Got it. Ok. We can do it later anyways.